eda/src/autogluon/eda/analysis/interaction.py

```diff
@@ -109,3 +109,99 @@ def _fit(self, state: AnalysisState, args: AnalysisState, **fit_kwargs) -> None:
+        variable to use as hue in x/y-analysis.
+    key: Optional[str], default = None
+        key to use to store the analysis in the state; the value is later to be used by FeatureInteractionVisualization.
+        If the key is not provided, then use one of theform: 'x:A|y:B|hue:C' (omit corresponding x/y/hue if the value not provided)
+        See also :py:class:`~autogluon.eda.visualization.interaction.FeatureInteractionVisualization`
+    parent: Optional[AbstractAnalysis], default = None
+        parent Analysis
+    children: Optional[List[AbstractAnalysis]], default None
+        wrapped analyses; these will receive sampled `args` during `fit` call
+    kwargs
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import numpy as np
+    >>> import autogluon.eda.analysis as eda
+    >>> import autogluon.eda.visualization as viz
+    >>> import autogluon.eda.auto as auto
+    >>>
+    >>> df_train = pd.DataFrame(...)
+    >>>
+    >>> state = auto.analyze(
+    >>>     train_data=df_train, label='Survived',
+    >>>     anlz_facets=[
+    >>>         eda.dataset.RawTypesAnalysis(),
+    >>>         eda.interaction.FeatureInteraction(key='target_col', x='Survived'),
+    >>>         eda.interaction.FeatureInteraction(key='target_col_vs_age', x='Survived', y='Age')
+    >>>     ],
+    >>>     viz_facets=[
+    >>>         # Bar Plot with counts per each of the values in Survived
+    >>>         viz.interaction.FeatureInteractionVisualization(key='target_col', headers=True),
+    >>>         # Box Plot Survived vs Age
+    >>>         viz.interaction.FeatureInteractionVisualization(key='target_col_vs_age', headers=True),
+    >>>     ]
+    >>> )
+    >>>
+    >>> # Simplified shortcut for interactions: scatter plot of Fare vs Age colored based on Survived values.
+    >>> auto.analyze_interaction(x='Fare', y='Age', hue='Survived', train_data=df_train)
+    """
+
+    def __init__(
+        self,
+        x: Optional[str] = None,
+        y: Optional[str] = None,
+        hue: Optional[str] = None,
+        key: Optional[str] = None,
+        parent: Optional[AbstractAnalysis] = None,
+        children: Optional[List[AbstractAnalysis]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(parent, children, **kwargs)
+        self.x = x
+        self.y = y
+        self.hue = hue
+        self.key = key
+
+    def can_handle(self, state: AnalysisState, args: AnalysisState) -> bool:
+        return self.all_keys_must_be_present(state, "raw_type")
+
+    def _fit(self, state: AnalysisState, args: AnalysisState, **fit_kwargs):
+        cols = {
+            "x": self.x,
+            "y": self.y,
+            "hue": self.hue,
+        }
+
+        if self.key is None:
+            # if key is not provided, then convert to form: 'x:A|y:B|hue:C'; if values is not provided, then skip the value
+            self.key = "|".join([f"{k}:{v}" for k, v in {k: cols[k] for k in cols.keys()}.items() if v is not None])
```

Comment:
I know this is very pythonic, but from my perspective this line of code is complicated enough to make people hard to understand. Shall we expand this into its regular form?

Reply:
Expanded